homeassistant/components/light/flux_led.py

```diff
@@ -247,10 +257,20 @@ def turn_on(self, **kwargs):
         if rgb is None:
             rgb = self._bulb.getRgb()
 
-        self._bulb.setRgb(*tuple(rgb), brightness=brightness)
+        if white is None and self._mode == MODE_RGBW:
+            white = self.white_value
 
-        if white is not None:
-            self._bulb.setWarmWhite255(white)
+        # handle W only mode (use brightness instead of white value)
+        if self._mode == MODE_WHITE:
+            self._bulb.setRgbw(0, 0, 0, w=brightness)
+
+        # handle RGBW mode
+        elif self._mode == MODE_RGBW:
+            self._bulb.setRgbw(*tuple(rgb), w=white, brightness=brightness)
+
+        # handle RGB mode
+        else:
+            self._bulb.setRgb(*tuple(rgb), brightness=brightness)
```

Comment:
Could you confirm that calling turn_on(rgb=[5,0,0], brighness=None) will set the RGB color to [5,0,0] and not to [255,0,0]  or anything else deformed by the previous brightness settings you re-use here ?

Reply:
Finally had a chance to test this and I found some inconsistent behavior.  However, it appears that it is not with the handling of the values by the flux component, but some issue with the way the values are passed into `turn_on`.

I used the following payload to test:
```
{
  "entity_id": "light.test_led_1",
  "rgb_color": [0,5,0]
}
```

`rgb_color` is being converted to `hs_color` before being passed into the `turn_on` method.  The flux component then uses `color_util.color_hs_to_RGB` to convert into a tuple of rgb values.  This conversion results in `0,255,0` for the payload above instead of the expected `0,5,0`.  This is the source of the inconsistent behavior that I'm experiencing.

The flux library will only recalculate rgb values if a numeric value is passed for brightness.  If `None` is passed, then it does not recalculate.  I am making a change so that existing brightness value is only passed if `rgb is None`.  This should prevent unexpected recalculation of the brightness when brightness is not included on the payload.

I'm not familiar enough with HA internals to know where to fix the hs_color bug mentioned above.

Reply:
@oblogic7: Thanks for investigating furhter ! 
It seems that this breakage that changes rgb_color [0,5,0] to [0,255,0] has been introduced in PR #11288. 
@armills: any clues/suggestions how to fix that ?

Reply:
After digging through the discussion on https://github.com/home-assistant/home-assistant/pull/11288, I see that @amelchio calls out this bug specifically.

https://github.com/home-assistant/home-assistant/pull/11288/files/1ee6a9f786c0508a7079aa2eaf282e03b2b83c9d#r175284144

I'm still checking out the changes, but I don't think the fix will be implemented on components unless there have been updates since that PR was merged.  Maybe a new method has been added that I haven't found yet?   If not, it looks to me like it will require more changes to the conversion methods.

Reply:
Popping in because I was tagged. I did not read all of these comments and I don't have flux_led hardware. Still, if you have specific questions about how things should work, I might be able to help.

Reply:
@amelchio Are you aware of a fix or work around for the issue that you point out here: https://github.com/home-assistant/home-assistant/pull/11288/files/1ee6a9f786c0508a7079aa2eaf282e03b2b83c9d#r175284144

If not, do you know if it will be a component level change or if it will be more involved?

Reply:
Great.  That is how it was implemented.  I will roll back the last change so that the current brightness is used. Thanks!